server/channels/api4/post.go

```diff
@@ -128,6 +128,20 @@ func createPost(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusCreated)
 
 	// Note that rp has already had PreparePostForClient called on it by App.CreatePost
+	// For burn-on-read posts, the author should see the revealed content in the API response
+	// to avoid relying on websocket events which may fail due to connection issues
+	if rp.Type == model.PostTypeBurnOnRead && rp.UserId == c.AppContext.Session().UserId {
+		revealedPost, appErr := c.App.GetSinglePost(c.AppContext, rp.Id, false)
+		if appErr != nil {
+			c.Err = appErr
+			return
+		}
+		// PreparePostForClient will reveal the post for the author
```

Comment:
This comment is incorrect as its is `GetSinglePost` that reveals the post, not `PreparePostForClient`